src/serac/physics/solid_functional.hpp

```diff
@@ -130,28 +135,34 @@ class SolidFunctional<order, dim, Parameters<parameter_space...>, std::integer_s
     state_.push_back(displacement_);
     state_.push_back(adjoint_displacement_);
 
+    if (calc_shape_ == ShapeDisplacement::On) {
+      state_.push_back(shape_displacement_);
+    }
+
     // Create a pack of the primal field and parameter finite element spaces
-    mfem::ParFiniteElementSpace*                                             test_space = &displacement_.space();
-    std::array<mfem::ParFiniteElementSpace*, sizeof...(parameter_space) + 2> trial_spaces;
+    mfem::ParFiniteElementSpace* test_space = &displacement_.space();
+
+    std::array<mfem::ParFiniteElementSpace*, 3 + sizeof...(parameter_space)> trial_spaces;
     trial_spaces[0] = &displacement_.space();
     trial_spaces[1] = &displacement_.space();
+    trial_spaces[2] = &displacement_.space();
 
     if constexpr (sizeof...(parameter_space) > 0) {
       tuple<parameter_space...> types{};
       for_constexpr<sizeof...(parameter_space)>([&](auto i) {
-        trial_spaces[i + 2] =
+        trial_spaces[i + 3] =
             generateParFiniteElementSpace<typename std::remove_reference<decltype(get<i>(types))>::type>(&mesh_);
         parameter_sensitivities_[i] = std::make_unique<FiniteElementDual>(mesh_, *trial_spaces[i + 2]);
       });
     }
 
-    residual_ = std::make_unique<Functional<test(trial, trial, parameter_space...)>>(test_space, trial_spaces);
-
-    state_.push_back(velocity_);
-    state_.push_back(displacement_);
+    residual_ = std::make_unique<Functional<test(trial, trial, trial, parameter_space...)>>(test_space, trial_spaces);
 
-    displacement_ = 0.0;
-    velocity_     = 0.0;
+    displacement_         = 0.0;
+    velocity_             = 0.0;
+    shape_displacement_   = 0.0;
+    adjoint_displacement_ = 0.0;
+    shape_displacement_   = 0.0;
```

Comment:
Is there a reason that `shape_displacement_` needs to be zeroed out both before and after `adjoint_displacement_`? I'm guessing no, but if so, a comment is in order.

Reply:
Nope. This is a typo.